crates/trigger/src/lib.rs

```diff
@@ -243,12 +243,14 @@ impl<Executor: TriggerExecutor> TriggerAppEngine<Executor> {
         let mut component_instance_pres = HashMap::default();
         for component in app.borrowed().components() {
             let id = component.id();
-            component_instance_pres.insert(
-                id.to_owned(),
-                Executor::instantiate_pre(&engine, &component, trigger_configs.get(id).unwrap())
-                    .await
-                    .with_context(|| format!("Failed to instantiate component '{id}'"))?,
-            );
+            if let Some(config) = trigger_configs.get(id) {
```

Comment:
It might be good to log a warning if the ID is _not_ found.  At the moment, dangling components are not an expected thing (although this may change with @fibonacci1729's composition work, I'm not when things get composed and what ends up in the lockfile), and I'd be inclined to give the user some way to find out that a surprising thing has happened.

Reply:
Sure